### PR TITLE
Update vueuse & ensure composables are only used inside `setup`

### PR DESCRIPTION
**Pull request by MetRonnie**
Supersedes / unblocks #1620

- Ensure the default view loaded from local storage is valid else fall back to tree view

- #1620 

- Ensure composables are only used synchronously[^1] inside `setup`

    > The vue docs say:
    > 
    > ...
    > 
    > > [Usage restrictions](https://vuejs.org/guide/reusability/composables.html#usage-restrictions)
    > > Composables should only be called **synchronously in <script setup> or the setup() hook**.

    https://github.com/vueuse/vueuse/issues/2608#issuecomment-1464196417

[^1]: Synchronous means not inside callbacks of e.g. computed properties, lifecycle hooks etc.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Tests are included
- [x] No changelog entry needed as not user-facing

### CODE REVIEW

File: src/components/cylc/analysis/BoxPlot.vue [hunk 1]
```diff
@@ -152,71 +222,6 @@ export default {
         { title: `Max ${this.timingOption} time`, value: `max${upperFirst(this.timingOption)}Time` },
       ]
     },
-
-    chartOptions () {
-      return {
-        chart: {
-          animations: {
-            enabled: useReducedAnimation().value ? false : this.animate,
```
Comment (oliver-sanders) [inline src/components/cylc/analysis/BoxPlot.vue:160]: Do we really have to shift this whole block just to handle this one usage?

**Comment by markgrahamdawson** [comment]
https://vueschool.io/lessons/calling-composables-in-the-wrong-place

File: src/components/cylc/analysis/BoxPlot.vue (hunk 1 above)
Comment (MetRonnie) [reply src/components/cylc/analysis/BoxPlot.vue:160]: Yes

File: src/components/cylc/analysis/BoxPlot.vue [hunk 2]
```diff
@@ -97,6 +98,75 @@ export default {
     },
   },
 
+  setup (props) {
+    const reducedAnimation = useReducedAnimation()
+
+    const chartOptions = computed(() => ({
```
Comment (oliver-sanders) [inline src/components/cylc/analysis/BoxPlot.vue:104]: @cylc/data-workflows 

Heads up, this is just a different way of defining a `computed` property which is more kosher with the use of "composables" (aka things that start with the prefix `use`). No further changes required.

File: src/components/cylc/analysis/BoxPlot.vue (hunk 1 above)
Comment (oliver-sanders) [reply src/components/cylc/analysis/BoxPlot.vue:160]: Dang.

**Review by oliver-sanders — APPROVED** [review]

**Review by markgrahamdawson — APPROVED** [review]

**Pull request merged**